packages/atclient/src/metadata.c

```diff
@@ -1210,112 +1211,114 @@ int atclient_atkey_metadata_to_protocol_str(const atclient_atkey_metadata *metad
   memset(*metadata_str, 0, sizeof(char) * metadata_str_size);
 
   if (atclient_atkey_metadata_is_ttl_initialized(metadata)) {
-    sprintf(*metadata_str + pos, ":ttl:%ld", metadata->ttl);
-    pos += 5 + atclient_string_utils_long_strlen(metadata->ttl);
+    sprintf((*metadata_str) + pos, ":ttl:%ld", metadata->ttl);
+    pos += atclient_atkey_metadata_ttl_strlen(metadata);
   }
 
   if (atclient_atkey_metadata_is_ttb_initialized(metadata)) {
-    sprintf(*metadata_str + pos, ":ttb:%ld", metadata->ttb);
-    pos += 5 + atclient_string_utils_long_strlen(metadata->ttb);
+    sprintf((*metadata_str) + pos, ":ttb:%ld", metadata->ttb);
+    pos += atclient_atkey_metadata_ttb_strlen(metadata);
   }
 
   if (atclient_atkey_metadata_is_ttr_initialized(metadata)) {
-    sprintf(*metadata_str + pos, ":ttr:%ld", metadata->ttr);
-    pos += 5 + atclient_string_utils_long_strlen(metadata->ttr);
+    sprintf((*metadata_str) + pos, ":ttr:%ld", metadata->ttr);
+    pos += atclient_atkey_metadata_ttr_strlen(metadata);
   }
 
   if (atclient_atkey_metadata_is_ccd_initialized(metadata)) {
     if (metadata->ccd) {
-      sprintf(*metadata_str + pos, ":ccd:true");
-      pos += 9;
+      sprintf((*metadata_str) + pos, ":ccd:true");
     } else {
-      sprintf(*metadata_str + pos, ":ccd:false");
-      pos += 10;
+      sprintf((*metadata_str) + pos, ":ccd:false");
     }
+    pos += atclient_atkey_metadata_ccd_strlen(metadata);
```

Comment:
pattern 2: we implicitly call this same if statement inside `atclient_atkey_metadata_<thing>_strlen` for all boolean metadata, so we can consolidate the pos calculation by moving it outside the if/else